src/test/java/ay2021s1_cs2103_w16_3/finesse/model/transaction/CalculatedAmountTest.java

```diff
@@ -0,0 +1,66 @@
+package ay2021s1_cs2103_w16_3.finesse.model.transaction;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.math.BigDecimal;
+
+import org.junit.jupiter.api.Test;
+
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.Amount.CalculatedAmount;
+
+public class CalculatedAmountTest {
+
+    private CalculatedAmount zero = new CalculatedAmount();
+    private CalculatedAmount positive = new CalculatedAmount(new Amount("1"));
+    private CalculatedAmount negative = zero.subtract(positive);
+
+    @Test
+    public void zeroValue() {
+        assertEquals(BigDecimal.ZERO, zero.getValue());
+        assertTrue(zero.isNonNegative());
+    }
+
+    @Test
+    public void isNonNegative() {
```

Comment:
Maybe add:
```
        assertTrue(zero.isNonNegative());
```
Since it is a boundary value.

Reply:
I've moved it from the `zeroValue` test, it should make more sense this way.